pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -221,6 +221,9 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 					defer cancel()
 				}
 			}
+			if s.Env.StatusReporter != nil {
+				go s.Env.StatusReporter.RegisterEvent(con.ConID, discReq.TypeUrl, discReq.ResponseNonce)
```

Comment:
Why a go routine here? It seems like it should be the responsibility of `RegisterEvent` to kick off a goroutine if its needed, but looking at the implementation I am not sure if its needed at all

Reply:
My goal is to ensure that the K8s->Envoy data pipeline is not impacted by this change.  Particularly since the RegisterEvent function requests a lock, I don't want to block Pilot's config processing pipeline for the purpose of status.  If you'd like me to move the goroutine into the function, I'm happy to do so.

Reply:
You will create a TON of go routines.

Reply:
In particular if something is wrong or a push happens.

Reply:
I know goroutines are pretty scalable... up to 300,000 on an average processor if I recall.  Are we talking more than that?  These goroutines should be very short lived unless there is lock contention.  If we find that we are experiencing a lot of lock contention, I can implement some improved, non-blocking data flows here.

Reply:
Memory allocation is also pretty scalable - yet we spend all CPU in GC.
There are legitimate uses of goroutines - this doesn't seem to be one. If RegisterEvent is doing a blocking operation - maybe use a queue and have a single go-routine execute operations.

Reply:
Implemented non-blocking queue with single reader goroutine, removed 'go' prefixes in ads.